manifests/init.pp

```diff
@@ -34,7 +34,7 @@
   $write_threads                                                      = $collectd::params::write_threads,
 ) inherits collectd::params {
 
-  $collectd_version_real = pick($::collectd_version, $minimum_version)
+  $collectd_version_real = pick_default($facts['collectd_version'], $minimum_version)
```

Comment:
I'm not sure if `minimum_version` is allowed to be set to `undef`, but `pick_default` would allow this.